src/Moq/MethodCall.cs

```diff
@@ -123,6 +123,11 @@ public override void Execute(Invocation invocation)
 
 		public virtual void SetCallBaseResponse()
 		{
+			if (typeof(Delegate).IsAssignableFrom(this.Mock.TargetType))
```

Comment:
Moq defines an [extension method `Type.IsDelegate()`](https://github.com/moq/moq4/blob/2c241fbd4599da0069668d07f69441d66a9f8687/src/Moq/Extensions.cs#L75-L81), could you please use it here?